packages/@aws-cdk/aws-synthetics/test/code.test.ts

```diff
@@ -0,0 +1,124 @@
+import '@aws-cdk/assert/jest';
+import * as path from 'path';
+import { App, Stack } from '@aws-cdk/core';
+import * as synthetics from '../lib';
+
+let stack: Stack;
+let app: App;
+beforeEach(() => {
+  app = new App();
+  stack = new Stack(app, 'canaries');
+});
+
+describe('Code.fromInline', () => {
+  test('basic use case', () => {
+    // WHEN
+    new synthetics.Canary(stack, 'Canary', {
+      test: synthetics.Test.custom({
+        handler: 'index.handler',
+        code: synthetics.Code.fromInline('exports.handler = async () => {\nconsole.log(\'hello world\');\n};'),
+      }),
+    });
+
+    // THEN
+    expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+      Code: {
+        Handler: 'index.handler',
+        Script: 'exports.handler = async () => {\nconsole.log(\'hello world\');\n};',
+      },
+    });
+  });
+
+  test('fails if size exceeds 460800 bytes', () => {
+    expect(() => new synthetics.Canary(stack, 'Canary', {
+      test: synthetics.Test.custom({
+        handler: 'index.handler',
+        code: synthetics.Code.fromInline(generateRandomString(460801)),
+      }),
+    }))
+      .toThrowError('Canary source is too large, must be <= 460800 but is 460801');
```

Comment:
Also, that is the ONE case where you should **not** include the failing string verbatim (it'd be too large!!)